File: GrowthWare_VB/GrowthWareFramework/DataAccessLayer/Interfaces/IDirectories.vb
Imports GrowthWare.Framework.DataAccessLayer.Interfaces.Base
Imports GrowthWare.Framework.ModelObjects

Namespace DataAccessLayer
	Public Interface IDirectories
		Inherits IDBInteraction

		Property Profile() As MDirectoryProfile

		Function GetDirectories() As DataTable
		Function SaveDirectory() As Boolean
		Function AddDirectory() As Integer
	End Interface
End Namespace

File: Net_Core/DataAccess/Interfaces/IDirectories.cs

using GrowthWare.DataAccess.Interfaces.Base;
using GrowthWare.Framework.Models;
using System.Data;
using System.Threading.Tasks;

namespace GrowthWare.DataAccess.Interfaces;

/// <summary>
/// Interface IDDirectories
/// </summary>
public interface IDirectories : IDBInteraction
{
	/// <summary>
	/// Gets the directories.
	/// </summary>
	/// <returns>DataTable.</returns>
	Task<DataTable> Directories();

	/// <summary>
	/// Saves the specified profile.
	/// </summary>
	/// <param name="profile">The profile.</param>
	Task Save(MDirectoryProfile profile);

	/// <summary>
	/// Gets or sets the security entity seq ID.
	/// </summary>
	/// <value>The security entity seq ID.</value>
	int SecurityEntitySeqId { get; set; }
}